### PR TITLE
chore: add custom sync repo settings

### PR DESCRIPTION
**Pull request by parthea**

### CODE REVIEW

**Comment by busunkim96** [comment]
@parthea There seem to be samples in this repo: https://source.cloud.google.com/results/invocations/028324d7-0c47-4143-8084-7d97d28de548/targets/github%2Fpython-bigquery-sqlalchemy%2Fsamples%2Fsnippets/tests

**Comment by shollyman** [comment]
Deferring LGTM on this one to swast, I'm not sure if the sqlalchemy workflows needs any customization.

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Defaults to `main`
+- pattern: main
+  requiresCodeOwnerReviews: true
+  requiresStrictStatusChecks: true
+  requiredStatusCheckContexts:
+    - 'cla/google'
+    - 'OwlBot Post Processor'
+    - 'Kokoro'
+    - 'Kokoro SQLAlchemy compliance'
+    - 'Kokoro prerelease'
```
Comment (tswast) [inline .github/sync-repo-settings.yaml:14]: Let's not block PR merges due to prerelease failures. I do want to know about them and get flakeybot issues, though.

File: .github/sync-repo-settings.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Identifies the protection rule pattern. Name of the branch to be protected.
+# Defaults to `main`
+- pattern: main
+  requiresCodeOwnerReviews: true
+  requiresStrictStatusChecks: true
+  requiredStatusCheckContexts:
+    - 'cla/google'
+    - 'OwlBot Post Processor'
+    - 'Kokoro'
+    - 'Kokoro SQLAlchemy compliance'
```
Comment (tswast) [inline .github/sync-repo-settings.yaml:13]: I'm undecided as to whether we should block PRs for this. They can take hours to run and we don't have direct control over them, so I'm inclined to say they shouldn't be blocking (but if they're broken we should fast-follow with a fix)

Comment (tswast) [reply .github/sync-repo-settings.yaml:13]: Side note: it'd be lovely if we could have this block release PRs only.

**Review by tswast — CHANGES_REQUESTED** [review]

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (tswast) [inline .github/sync-repo-settings.yaml:14]: Using a suggested commit so I can resolve my previous comments.

```suggestion
```

**Review by tswast — APPROVED** [review]

**Comment by tswast** [comment]
@parthea I think this will require an admin to merge due to missing GH Actions tests.

**Comment by tswast** [comment]
Note: Failing Kokoro tests should be fixed by #401 , but that's blocked waiting on review and also this change to GH Actions requirements.

**Pull request merged**